plugins/hls-ormolu-plugin/src/Ide/Plugin/Ormolu.hs

```diff
@@ -44,13 +47,26 @@ provider ideState typ contents fp _ = withIndefiniteProgress title Cancellable $
     fullRegion = RegionIndices Nothing Nothing
     rangeRegion s e = RegionIndices (Just $ s + 1) (Just $ e + 1)
     mkConf o region = defaultConfig { cfgDynOptions = o, cfgRegion = region }
-    fmt :: T.Text -> Config RegionIndices -> IO (Either OrmoluException T.Text)
-    fmt cont conf =
+    fmt :: T.Text -> Config RegionIndices -> IO (Either SomeException T.Text)
+    fmt cont conf = flip catches handlers $ do
+      let fp' = fromNormalizedFilePath fp
 #if MIN_VERSION_ormolu(0,5,3)
-      try @OrmoluException $ ormolu conf (fromNormalizedFilePath fp) cont
+      cabalInfo <- getCabalInfoForSourceFile fp' <&> \case
+        CabalNotFound                -> Nothing
+        CabalDidNotMention cabalInfo -> Just cabalInfo
+        CabalFound cabalInfo         -> Just cabalInfo
```

Comment:
Since this function is not using the information from hie-bios, we are destined to have unexpected behaviour with this at some point in the future when ghcide and ormolu don't agree on the .cabal file. This seems to be loading information from .cabal files, without knowing which .cabal file is the correct one?
I am uncertain whether this is really better, but we could ask for the cradle options for this file in particular and find the "more correct" .cabal file if it is a stack or cabal project. Do you know what the cabal file used for?

Reply:
The idea is that the Ormolu invoked via HLS should have equal behavior with the Ormolu executable (this currently being not the case is the point of #3448). For how the information from the `.cabal` file is used by Ormolu, see this section in its README: https://github.com/tweag/ormolu#language-extensions-dependencies-and-fixities

More generally, this touches #411; i.e. in the distant future, it could be the case that all formatters are invoked via HLS, then using hie-bios etc.

Reply:
You are right, this is more about #411